src/argilla_sdk/settings/_resource.py

```diff
@@ -392,13 +345,14 @@ def _validate_empty_settings(self):
     def _validate_duplicate_names(self) -> None:
         dataset_properties_by_name = {}
 
-        for prop in self.fields + self.questions + self.vectors + self.metadata:
-            if prop.name in dataset_properties_by_name:
-                raise SettingsError(
-                    f"names of dataset settings must be unique, "
-                    f"but the name {prop.name!r} is used by {type(prop).__name__!r} and {type(dataset_properties_by_name[prop.name]).__name__!r} "
-                )
-            dataset_properties_by_name[prop.name] = prop
+        for properties in [self.fields, self.questions, self.vectors, self.metadata]:
```

Comment:
This change is include to avoid implement the `__add__` method to support `self.fields + self.vectors...`